src/server/editorServices.ts

```diff
@@ -1589,6 +1589,8 @@ namespace ts.server {
                 configFileErrors.push(...parsedCommandLine.errors);
             }
 
+            project.setTypeAcquisition(parsedCommandLine.typeAcquisition!); // TODO: GH#18217
+

```

Comment:
I think instead of this, you want to move `const filesToAdd = parsedCommandLine.fileNames.concat(project.getExternalFiles());` to `updateNonInferredProjectFiles` and 

In `addFilesToNonInferredProject` you want to move `project.setTypeAcquisition(typeAcquisition);` before calling `updateNonInferredProjectFiles`